SignalServiceKit/src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -364,6 +364,10 @@ - (void)fallbackTimerDidFire
     }
 
     dispatch_async(OWSDisappearingMessagesJob.serialQueue, ^{
+        if (!CurrentAppContext().isMainAppAndActive) {
```

Comment:
I thought this was easier to reason about than invalidating/restarting the timer on app activation/deactivation.

Reply:
Maybe we should (crudely) "time" the perf of this work using NSDate or something and do "verbose" (or higher) logging if it takes more than 5 seconds?